### ISSUE TITLE
[Panel] Improve accessibility of page status

### ISSUE THREAD

**Issue by LeBenLeBen** [getkirby/kirby#1444]
**Describe the bug**
Page status dot doesn't provide textual status of the page, only visual clue with colors.

**To Reproduce**
1. Keep your mouse above a page status dot for about a second
2. See the "title" displayed as "Status" whatever the status of the page is

**Expected behavior**
The title should mention the actual status of the page, like: "Status: draft".

**Kirby Version**
3.0.0

**Comment by distantnative** [comment]
@LeBenLeBen To give you some insight why this hasn't been the trivial thing as we first thought it to be:
Every page blueprint can define custom status names/labels. With pages with different blueprints in the list, we would need to get the information on this for each page to display the correct status name instead of the three default labels.

**Comment by ralfgoeke** [comment]
Just stumbled across this issue as well while working with a color blind client. Maybe we can make this at least a tiny bit more accessible, by adding the dot color in the tooltip at least?

`aria-label="Status: red"`

What do you think?

**Comment by sebastiangreger** [comment]
I've been thinking about this issue as well; happy to see this ticket :)

Since color alone [should](https://www.w3.org/TR/WCAG21/#use-of-color) never be used as the only visual means to convey information, I've been thinking along the lines of enhancing the "dot" with a symbol.

While getting the markup and title attribute fixed is important, could the solution also cover keeping the current colour dots (they work great for users not affected by color blindness), but an additional black (or otherwise high contrast) icon on top of it indicates the status?
e.g. ⊝ draft, ⊙ unlisted, ⊕ listed
or ⊝ draft, ⊗ unlisted, ⊕ listed

...or replace the color coding with a "moon" icon - an empty circle is a draft, a half-filled circle is unlisted, a filled circle is listed (if wanting to keep the current colors, it could be a red empty circle, a blue circle half filled in blue, or a filled green circle).

**Comment by distantnative** [comment]
I have moved this to 3.3.1 as we want to focus on feature progress in the major releases. However, I am adding also our `this release` flag for 3.3.1, in so far that we should really make up our minds for 3.3.1 if/how we should deal with this.

**Comment by distantnative** [comment]
I think we have to come to the conclusion that the full extent of this issue won't be solved in Kirby 3.
I suggest to add the standard labels (draft, listed, unlisted) as title/aria-label and then close it.

**Comment by sebastiangreger** [comment]
While I understand this decision from a product perspective, may I provide one more suggestion to improve this with very simple means?

Using `aria-label` and `title` comes with its own challenges (see e.g. [here](https://github.com/w3c/html/issues/1425#issuecomment-398402309)), with `title` even [discouraged](https://html.spec.whatwg.org/multipage/dom.html#the-title-attribute) by the WHATWG. While this duplicate attribute may indeed lead to a tooltip on some browsers, the proposed change is unlikely to improve usability/accessibility for users with limited colour vision at large; even less so on touchscreen devices where tooltips are not visible. And given that at least [4-5%](https://www.welt.de/kmpkt/article159112724/Wie-farbenblinde-Menschen-die-Welt-sehen.html) of Kirby users can be assumed to be affected by some form of lack of distinction between colours, I believe that this is an important enough issue.

As I pointed out earlier, the key to a solution is to not convey information by colour alone in the visual design. My earlier suggestions were only pointers and after some more consideration I do not believe that adding a symbol inside the circle would solve the problem, as -- I missed that part last time -- the icons also appear in a "non-filled" variant for pages where the user does not have the rights to change the status. Hence, this would be too complex (and it would be a rather arbitrary icon language).

Upon further deliberation, I would suggest to solve this accessibility issue by using different shapes:

![icons svg](https://user-images.githubusercontent.com/6355217/74921083-86038480-53cd-11ea-88e5-b86a8a24ad28.png)

My personal preference between these two would be the latter, but there might be other, yet unexplored, options as well. The "Stop sign" is a rather common shape in many (not all) cultures, and the rest could be somewhat derived from that (Stop sign: no, Square: visible page, Circle: shape in between). Adding these shapes does not reduce usability for anybody, but may make using the Kirby panel easier for a significant share of users. I of course acknowledge that it has a visual impact on the simplicity in appearance.

On a side note: for a fully WCAG compliant solution, also the colour contrast between the red, blue, and green hues used for this icon and the grey background would have to be reviewed, as they do not appear to meet the [criterion](https://www.w3.org/WAI/WCAG21/Understanding/non-text-contrast.html) of a 3:1 colour contrast. If a change to the colour scheme is not desired, this could for example be mitigated by adding a darker border to the icon.

**Comment by lukasbestle** [comment]
I like the idea to vary the shapes. The stop sign, circle and square variant looks pretty good.

**Comment by distantnative** [comment]
+1

**Comment by sebastiangreger** [comment]
Glad to hear!

I'm probably spending way more time on this than I should, but here are two more explorations, which would help avoid the mix of round and square shapes:

![icons2 svg](https://user-images.githubusercontent.com/6355217/74936566-a55cda80-53ea-11ea-9c77-11ee9d735139.png)

The circle variant is tricky; if it's a true half-circle it appears too small, but as a 3/4 circle like this it looks odd as well).

...I'll try to stop here :wink:

**Comment by distantnative** [comment]
The different shapes started looking really quirky and irritating/grabbing too much attention (it looked broken basically haha).

I came up with an alternative approach:
<img width="593" alt="Screen Shot 2020-03-07 at 12 44 23" src="https://user-images.githubusercontent.com/3788865/76142812-675de880-6071-11ea-8543-bd7cba8963ac.png">


Using the different circle shapes for the status and a small lock for when you can't change the status.

**Comment by distantnative** [comment]
✅ will be addressed with 3.4
